drivers/sensor/tdk/icm456xx/icm456xx.c

```diff
@@ -0,0 +1,1061 @@
+		if (drv_data->gyro_hz == 0) 
+			inv_imu_set_gyro_mode(&drv_data->driver, PWR_MGMT0_GYRO_MODE_LN);
+	} else if (val->val1 == 0) {
+		inv_imu_set_gyro_mode(&drv_data->driver, PWR_MGMT0_GYRO_MODE_OFF);
+		drv_data->gyro_hz = val->val1;
+	} else {
+		LOG_ERR("Incorrect sampling value");
+		return -EINVAL;
+	}
+	return 0;
+}
+
+static int icm456xx_set_gyro_fs(struct icm456xx_data *drv_data, const struct sensor_value *val)
+{
+	int32_t val_dps = sensor_rad_to_degrees(val);
+
+	if (val_dps > 2000 || val_dps < 15) {
+		LOG_ERR("Incorrect fullscale value");
+		return -EINVAL;
+	}
+	inv_imu_set_gyro_fsr(&drv_data->driver,
+			convert_gyr_fs_to_enum(val_dps, &drv_data->gyro_fs));
+	LOG_DBG("Set gyro fullscale to: %d dps", drv_data->gyro_fs);
+	return 0;
+}
+
+static int icm456xx_accel_config(struct icm456xx_data *drv_data, enum sensor_attribute attr,
+				 const struct sensor_value *val)
+{
+	if (attr == SENSOR_ATTR_CONFIGURATION) {
+		icm456xx_set_accel_power_mode(drv_data, val);
+
+	} else if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+		icm456xx_set_accel_odr(drv_data, val);
+
+	} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+		icm456xx_set_accel_fs(drv_data, val);
+
+	} else if ((enum sensor_attribute_icm456xx)attr == SENSOR_ATTR_BW_FILTER_LPF) {
+		if (val->val1 > 180) {
+			LOG_ERR("Incorrect low pass filter bandwidth value");
+			return -EINVAL;
+		}
+		inv_imu_set_accel_ln_bw(&drv_data->driver, convert_ln_bw_to_enum(val->val1));
+
+	} else if ((enum sensor_attribute_icm456xx)attr == SENSOR_ATTR_AVERAGING) {
+		if (val->val1 > 64 || val->val1 < 2) {
+			LOG_ERR("Incorrect averaging filter value");
+			return -EINVAL;
+		}
+		inv_imu_set_accel_lp_avg(&drv_data->driver, convert_lp_avg_to_enum(val->val1));
+	} else {
+		LOG_ERR("Unsupported attribute");
+		return -EINVAL;
+	}
+	return 0;
+}
+
+static int icm456xx_gyro_config(struct icm456xx_data *drv_data, enum sensor_attribute attr,
+				 const struct sensor_value *val)
+{
+	if (attr == SENSOR_ATTR_SAMPLING_FREQUENCY) {
+		icm456xx_set_gyro_odr(drv_data, val);
+
+	} else if (attr == SENSOR_ATTR_FULL_SCALE) {
+		icm456xx_set_gyro_fs(drv_data, val);
+	} else if ((enum sensor_attribute_icm456xx)attr == SENSOR_ATTR_BW_FILTER_LPF) {
+		if (val->val1 > 180) {
+			LOG_ERR("Incorrect low pass filter bandwidth value");
+			return -EINVAL;
+		}
+		inv_imu_set_gyro_ln_bw(&drv_data->driver, convert_ln_bw_to_enum(val->val1));
+	} else {
+		LOG_ERR("Unsupported attribute");
+		return -EINVAL;
+	}
+	return 0;
+}
+
+static int icm456xx_sensor_init(const struct device *dev)
+{
+	struct icm456xx_data *data = dev->data;
+	const struct icm456xx_config *config = dev->config;
+	int err = 0;
+
+	/* Initialize serial interface and device */
+	data->driver.transport.context = (struct device *)dev;
+	data->driver.transport.read_reg = inv_io_hal_read_reg;
+	data->driver.transport.write_reg = inv_io_hal_write_reg;
+	data->driver.transport.serif_type = config->serif_type;
+	data->driver.transport.sleep_us = inv_sleep_us;
+
```

Comment:
Why don't you call inv_imu_init API?

Reply:
Current eMD doesn't have init() functions for initialize transport API

Reply:
You can use inv_imu_adv_init()

Reply:
I changed all adv related APIs to normal driver API for code compatible of other eMD like 42x70.